MobiFlight/MobiFlightFirmwareUpdater.cs

```diff
@@ -38,31 +39,108 @@ public static bool Update(MobiFlightModule module)
             String Port = module.InitUploadAndReturnUploadPort();
             if (module.Connected) module.Disconnect();
 
-            while (!SerialPort.GetPortNames().Contains(Port))
-            {
-                System.Threading.Thread.Sleep(100);
-            }
-
             if (module.Board.AvrDudeSettings != null)
             {
-                try {
+                while (!SerialPort.GetPortNames().Contains(Port))

```

Comment:
This serial port searching only applies to avrdude flashing so I moved it inside the avrdude flash block.